src/d3a/models/area/__init__.py

```diff
@@ -109,6 +143,29 @@ def __init__(self, name: str = None, children: List["Area"] = None,
             if external_connection_available is True else None
         self.should_update_child_strategies = False
 
+    @property
+    def children(self):
+        return self.__children
+
+    @children.setter
+    def children(self, new_children: AreaChildrenList):
+        if type(new_children) == list:
+            self.__children = AreaChildrenList(self, new_children)
+        elif type(new_children) == AreaChildrenList:
+            new_children.parent_area = self
+            self.__children = new_children
+
+    @property
+    def name(self):
+        return self.__name
+
+    @name.setter
+    def name(self, new_name):
+        if not check_area_name_exists_in_parent_area(self.parent, new_name):
+            self.__name = new_name
+        else:
+            raise AreaException("Area name should be unique inside the same Parent Area")
```

Comment:
Minor comment: AFAIK we dont plan to have an area_renaming functionality, but leave it